### PR TITLE
Retrieve DSCs from server (EXPOSUREAPP-8362)

### PR DESCRIPTION
**Pull request by jurajkusnier**
implement `DscServer` response for downloading List of DSCs from server

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServer.kt [hunk 1]
```diff
@@ -1,12 +1,70 @@
...
+) {
 
-    suspend fun getDscList(): ByteArray {
-        throw NotImplementedError()
+    suspend fun getDscList(): DscList {
+        return try {
+            dscApi.dscList().parseAndValidate(
+                ErrorCode.FILE_MISSING,
+                ErrorCode.SIGNATURE_INVALID,
+                ErrorCode.EXTRACTION_FAILED
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServer.kt:28]: The error codes can be hardcoded.
This was passed in a different implementation as the parsing logic was re-used.
I think here the `dscList()` will be the only use, right?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServerModule.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+@Module
+class DscServerModule {
+
+    @Reusable
+    @Provides
+    fun apiV1(
+        @DownloadCDNHttpClient httpClient: OkHttpClient,
+        @DownloadCDNServerUrl url: String,
+        @CertificateValueSet cache: Cache,
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServerModule.kt:22]: Is this the right cache?

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServerModule.kt:22]: Did you mean `@CertificateValidation`?

**Review by d4rken — COMMENTED** [review]
Nice tests :heart:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServer.kt (hunk 1 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServer.kt]: updated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServerModule.kt (hunk 2 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/server/DscServerModule.kt]: updated

**Review by d4rken — APPROVED** [review]
lgtm

**Review by mtwalli — APPROVED** [review]
LGTM

**Pull request merged**